python/langsmith/_internal/_background_thread.py

```diff
@@ -30,6 +36,8 @@
 
 logger = logging.getLogger("langsmith.client")
 
+HTTP_REQUEST_THREAD_POOL = cf.ThreadPoolExecutor(max_workers=cpu_count() * 3)
```

Comment:
Should we delay creation of this until the first run?
Langsmith has a lot of downstream deps (through langchain)  that may not be using tracing


ALso, why cpu count * 3? Is the zstd C lib spinning up a separate process?

Reply:
The max workers is separate form zstd. The workers handle the requests to the backend. Since this is I/O bound it can be greater than the cpu count.

Reply:
An empty ThreadPoolExecutor is relatively lightweight and no actual threads are created until work is submitted. The benefits of lazy creating one seems minimal